### PR TITLE
Remove variant lists

### LINKED ISSUE #204: Deprecate VariantLists
The parameterized terms proposal (#176) supersedes `VariantLists`. Let's <del>remove them from the syntax before 1.0</del><ins>deprecate them in 0.8 and remove later before 1.0</ins>.

### PR DESCRIPTION
**Pull request by stasm**
Fix <del>#204</del><ins>#214</ins>.

### CODE REVIEW

**Comment by stasm** [comment]
This PR depends on #205. I'll wait for it to land before I request review. @Pike, I'm just putting this on your radar for now, in case it makes reviewing #205 easier.

File: spec/CHANGELOG.md [hunk 1]
```diff
@@ -66,7 +77,7 @@
     `VariantLists`:
 
     ```properties
-    # A Term with a VariantList as a value.
+    # BEFORE A Term with a VariantList as a value.
     -thing = {
        *[definite] the thing
        *[indefinite] a thing
```
Comment (Pike) [inline spec/CHANGELOG.md]: drive by nit, the double defaults are parsing errors :-)

File: test/fixtures/member_expressions.ftl [hunk 2]
```diff
@@ -1,6 +1,5 @@
-variant-expression = {-term[case]}
```
Comment (Pike) [inline test/fixtures/member_expressions.ftl]: Should we keep this as an error test?

File: test/fixtures/variant_keys.ftl [hunk 3]
```diff
@@ -1,37 +1,37 @@
--simple-identifier =
-    {
+simple-identifier =
+    { $sel ->
```
Comment (Pike) [inline test/fixtures/variant_keys.ftl]: Can you cross-check which of these are already tested in `select_expressions.ftl `?

Comment (stasm) [reply test/fixtures/variant_keys.ftl]: I checked. `select_expressions.ftl` is mostly about selectors and things like empty variant values. In `variant_keys.ftl` I'd like to test different variant keys and their errors. I made a few changes in `select_expressions.ftl` to simplify the variant keys used there (`key` or `one` and `two` for nested placeables).

File: test/fixtures/member_expressions.ftl (hunk 2 above)
Comment (stasm) [reply test/fixtures/member_expressions.ftl]: I added a new fixture file, called `obsolete.ftl`, with a few examples of discontinued syntax. Thanks for the suggestion.

**Pull request merged**